File: 2017/j3.py
# code by savir singh

start = input().split(" ")
end = input().split(" ")

power = int(input())
startx = int(start[0])
starty = int(start[1])
endx = int(end[0])
endy = int(end[1])

powerneeded = abs(endx-startx)+abs(endy-starty)
if power >= powerneeded:
    if powerneeded % 2 == 0 and power % 2 == 0:
        print("Y")
    elif powerneeded % 2 != 0 and power % 2 != 0:
        print("Y")
    else:
        print("N")
else:
    print("N")

File: 2017/j3.cpp
// code by savir singh

#include <bits/stdc++.h>
using namespace std;

int main() {
    int a, b, c, d, power;
    cin >> a >> b >> c >> d >> power;
    int power_needed = abs(c-a)+abs(d-b);
    if (power>=power_needed) {
        if (power_needed%2==0 && power%2==0) {
            cout << "Y";
        }
        else if (power_needed%2!=0 && power%2!=0) {
            cout << "Y";
        }
        else {
            cout << "N";
        }
    }
    else {
        cout << "N";
    }
}
